vaadinfrontend/src/main/java/life/qbic/datamanager/views/project/create/ProjectInformationDialog.java

```diff
@@ -216,6 +246,13 @@ public void reset() {
   private void resetChildValues(Component component) {
     component.getChildren().filter(comp -> comp instanceof HasValue<?, ?>)
         .forEach(comp -> ((HasValue<?, ?>) comp).clear());
+
+  }
+
+  private void defaultProjectCodeCreation() {
+    this.addOpenedChangeListener(opend -> {
+      codeField.setValue(ProjectCode.random().value());
+    });
```

Comment:
The opened change listener also fires on close.
```suggestion
  private void defaultProjectCodeCreation() {
    this.addOpenedChangeListener(opend -> 
    {
    if opened is window.opened // or similar
      codeField.setValue(ProjectCode.random().value());
    });
```